kpm/lib/kpm/database.rb

```diff
@@ -0,0 +1,78 @@
+
+module KPM
+
+  class Database
+    class << self
+
+      # Destination database
+      DATABASE = 'killbill'
+      USERNAME = 'root'
+      PASSWORD = 'root'
+
+      MYSQL_COMMAND_LINE = "mysql #{DATABASE} --user=#{USERNAME} --password=#{PASSWORD} -e"
+
+      @@mysql_command_line = MYSQL_COMMAND_LINE
+
+      def set_configuration(database = nil, user = nil, password = nil)
+
+        if database && user && password
+          @@mysql_command_line = "mysql #{database} --user=#{user} --password=#{password} -e"
+        end
+
+      end
+
+      def execute_insert_statement(table_name, query)
+        response = `#{@@mysql_command_line} "#{query}" 2>&1`
+
+        if response.include? 'ERROR'
+          puts "Importing #{table_name}...... \e[91;1m#{response}\e[0m"
+
+          return false
+        end
+
+        if response.include? 'LAST_INSERT_ID'
+          puts "\e[32mImporting #{table_name}...... success\e[0m"
+
+          return response.split("\n")[1]
+        end
+
+        return true
+      end
+
+      def generate_insert_statement(tables)
+
+        statements = Hash.new
+        puts "\e[32mGenerating statements\e[0m\n\n"
+        tables.each_key { |table_name|
+          rows = tables[table_name]
+          statements[table_name] = []
+
+          rows.each { |row|
+            columns = []
+            values = []
+
+            row.each_key { |name|
+              columns << name
+              values << row[name]
+            }
+
+            columns_names = columns.join(",").gsub(/'/,'')
+            value_data = values.map{|value| "'#{value}'"}.join(",")
+
+            statements[table_name] << get_insert_statement(table_name,columns_names,value_data)
+          }
+        }
+
+        statements
+
+      end
+
+      def get_insert_statement(table_name, columns_names, values)
+        return "INSERT INTO #{table_name} ( #{columns_names} ) VALUES ( #{values}); SELECT LAST_INSERT_ID();"
```

Comment:
Two thoughts:

* We would need a mechanism to display warnings too (see `SHOW WARNINGS` in the original script).
* I'm a bit concerned by the performance of generating single `INSERT` statements. We should at least do bulk loading. From https://dev.mysql.com/doc/refman/5.7/en/insert-optimization.html:

> If you are inserting many rows from the same client at the same time, use INSERT statements with multiple VALUES lists to insert several rows at a time. This is considerably faster (many times faster in some cases) than using separate single-row INSERT statements.

e.g. `INSERT INTO yourtable VALUES (1,2), (5,5), ...;`.

See also tips from https://dev.mysql.com/doc/refman/5.7/en/optimizing-innodb-bulk-data-loading.html .